File: BEEP/Common/Logging/Logger.cs
using System;

namespace Common.Logging
{
	public class Logger
	{

		string LogName;
		ILogTarget[] Targets;

		public Logger (string logName, ILogTarget[] targets)
		{
			this.LogName = logName;
			Targets = targets;
		}

		public void Log(string msg)
		{
			foreach (var target in Targets) {
				target.Log (string.Format("[{0}]: {1}", LogName, msg));
			}
		}

		public void Log(string msg, params object[] args)
		{
			msg = String.Format (msg, args);

			foreach (var target in Targets) {
				target.Log (string.Format("[{0}]: {1}", LogName, msg));
			}
		}

	}
}

File: Server/app/logging/Logger.scala
package logging

/**
 * Created by don on 1/24/15.
 */
class Logger (name:String, targets:List[ILogTarget]) {

  private def LogName:String = name

  def Log(msg:String) : Unit = {

    val output = "[%s::%s] %s".format("Log", LogName, msg)
    for(target <- targets ) target.Write(output)
  }

  def Error(msg:String) : Unit = {
    val output = "[%s::%s] %s".format("Error", LogName, msg)
    for(target <- targets ) target.Write(output)
  }

  def Info(msg:String) : Unit = {
    val output = "[%s::%s] %s".format("Info", LogName, msg)
    for(target <- targets ) target.Write(output)
  }

  def Warn(msg:String) : Unit = {
    val output = "[%s::%s] %s".format("Warn", LogName, msg)
    for(target <- targets ) target.Write(output)
  }

}
